hare/hare.go

```diff
@@ -328,10 +351,18 @@ func (h *Hare) outputCollectionLoop(ctx context.Context) {
 
 // listens to new layers.
 func (h *Hare) tickLoop(ctx context.Context) {
-	for {
+	for layer := h.layerClock.GetCurrentLayer(); ; layer++ {
 		select {
-		case layer := <-h.beginLayer:
-			go h.onTick(ctx, layer)
+		case <-h.layerClock.AwaitLayer(layer):
+			if h.layerClock.LayerToTime(layer).Sub(time.Now()) > (time.Duration(h.config.WakeupDelta) * time.Second) {
+				h.With().Warning("missed hare window, skipping layer", layer)
```

Comment:
this should be an error - it is very critical

Reply:
Under normal circumstances this can happen to regular users. For instance, if your computer sleeps and wakes up again - this warning will be printed for every layer that the node missed.

I think you're imagining the node running in a perfect environment (like a cloud node) and you're thinking that if the clock didn't tick on time then something terrible must have happened - but as I see it, this is something the node should be able to handle and recover from and it's only a problem when it happens unexpectedly. This is exactly what warnings are for, IMO: giving you a heads up when you try to investigate something, but communicating that it's not necessarily an issue.

---

Having said that, if you feel that this being a warning will make your life harder when investigating - I'll change it. I think that at some point we must decide (as a team) on a coherent strategy for what logs should belong to each level and then change the existing logs to match that (not a small task).